### DISCUSSION TITLE
How does one use Cookies

### DISCUSSION THREAD

**Discussion by MrMeapz** [jchristn/WatsonWebsocket#75] [Q&A] [answered]
Hello,

First of all. Awesome library. Really like what you've done here.

My question is how does one use cookies. What I'm trying to do is have my client connect to the server with a cookie attached to it, with for example a name and value. I then want the server to be able to grab this cookie from the connecting client by name and read the value of this cookie. I tried using args.HttpRequest.Cookies but it doesn't matter what I've done so far because it will always return an empty string.

I could not find anything about this in the documentation so I decided to ask here.

Any help would be highly appreciated.

**Comment by jchristn** [comment, marked as answer]
Hi @MeapVR thank you for your kind words, it is much appreciated.

I was able to do this using the ```Test.Server``` and ```Test.Client``` projects:
![image](https://user-images.githubusercontent.com/1421081/139545530-a5a38d6e-c6ea-4e46-bc33-fda315b0b9e7.png)

Specifically in the client, within ```InitializeClient```:
```
// before any .Start method
_Client.AddCookie(new System.Net.Cookie("foo", "bar", "/", "localhost"));
```

And within the server ```ClientConnected```:
```
if (args.HttpRequest.Cookies != null && args.HttpRequest.Cookies.Count > 0)
{
    foreach (Cookie cookie in args.HttpRequest.Cookies)
    {
        Console.WriteLine(cookie.Name + ": " + cookie.Value);
    }
}
```

Is that what you're attempting to do?

**Reply by MrMeapz** [reply]
Thanks for your fast reply.

This is exactly what I was looking for. Thank you very much. All I had to do was change localhost to 127.0.0.1 in your answer for testing locally in this case and then it worked like a charm!

Have a wonderful day!

**Reply by jchristn** [reply]
Enjoy!
